fix(doctor): fixes ArrayLiteralExpression error when running doctor with unmet peer dependency

benatshippabo opened the pull request:
**What's the problem this PR addresses?**
<!-- Describe the rationale of your PR. -->
<!-- Link all issues that it closes. (Closes/Resolves #xxxx.) -->

Fixes #2752 
...

**How did you fix it?**
<!-- A detailed description of your implementation. -->

Fixed a bad parameter when using `createSourceFile`
...

**Checklist**
<!--- Don't worry if you miss something, chores are automatically tested. -->
<!--- This checklist exists to help you remember doing the chores when you submit a PR. -->
<!--- Put an `x` in all the boxes that apply. -->
- [x] I have read the [Contributing Guide](https://yarnpkg.com/advanced/contributing).

<!-- See https://yarnpkg.com/advanced/contributing#preparing-your-pr-to-be-released for more details. -->
<!-- Check with `yarn version check` and fix with `yarn version check -i` -->
- [x] I have set the packages that need to be released for my changes to be effective.

<!-- The "Testing chores" workflow validates that your PR follows our guidelines. -->
<!-- If it doesn't pass, click on it to see details as to what your PR might be missing. -->
- [x] I will check that all automated PR checks pass before the PR gets reviewed.

arcanis commented:
It was intended, since `{"a": "b"}` is parsed differently in JS/TS than in JSON (a Block rather than an Expression).

benatshippabo commented:
> It was intended, since `{"a": "b"}` is parsed differently in JS/TS than in JSON (a Block rather than an Expression).

Okay I changed the script target to`ts.ScriptTarget.JSON` which should parse the file properly according to the source code:
https://github.com/microsoft/TypeScript/blob/1294ee8c1f37c8bf214db67e180edb3220a52236/src/compiler/parser.ts#L631-L633. 

I'm wondering why it didn't crash when `buildJsonNode` was trying to parse the json file after I removed the parentheses now 🤔

arcanis commented on packages/yarnpkg-doctor/sources/cli.ts:
```diff
@@ -192,8 +192,8 @@ async function buildJsonNode(p: PortablePath, accesses: Array<string>) {
-    `(${content})`,
-    ts.ScriptTarget.ES2015,
+    content,
+    ts.ScriptTarget.JSON,
```
Completely missed this parse mode 😅
